src/krate.rs

```diff
@@ -1189,11 +1189,13 @@ pub fn following(req: &mut Request) -> CargoResult<Response> {
 // this information already, but ember is definitely requesting it
 pub fn versions(req: &mut Request) -> CargoResult<Response> {
     let crate_name = &req.params()["crate_id"];
-    let tx = req.tx()?;
-    let krate = Crate::find_by_name(tx, crate_name)?;
-    let versions = krate.versions(tx)?;
-    let versions = versions.into_iter().map(|v| v.encodable(crate_name))
-                           .collect();
+    let conn = req.db_conn()?;
+    let krate = Crate::by_name(crate_name).first::<Crate>(&*conn)?;
+    let mut versions = Version::belonging_to(&krate).load::<Version>(&*conn)?;
+    versions.sort_by(|a, b| b.num.cmp(&a.num));
```

Comment:
This sorting is getting duplicated in multiple places-- do you think we could make a method that uses diesel that behaves like [`krate.versions()`](https://github.com/rust-lang/crates.io/blob/master/src/krate.rs#L568-L577) does now, that would always sort the versions? Or is there some reason why we wouldn't want that?